# Already fixed upstream, can be removed in the next integrate.
# https://github.com/triton-lang/triton/commit/2b2a872459648a7418d1c7e4cfa9aa8cb39e71e5

diff --git a/lib/Analysis/Utility.cpp b/lib/Analysis/Utility.cpp
--- a/lib/Analysis/Utility.cpp
+++ b/lib/Analysis/Utility.cpp
@@ -727,6 +727,10 @@ bool supportMMA(triton::DotOp op, int ve
       return false;
     if (op.getType().getRank() != 2)
       return false;
+    if (numWarps != 4 && numWarps != 8) {
+      // Currently only support numWarps 4 or 8 for TMEM load and store.
+      return false;
+    }
     if (!(numWarps % 4 == 0 && retShapePerCTA[rank - 2] % 64 == 0 &&
           retShapePerCTA[rank - 1] % 8 == 0))
       return false;
